app/scripts/admin/controllers/hiv-monthly-summary-indicators.controller.js

```diff
@@ -81,7 +82,12 @@
       if ($scope.countBy && $scope.countBy !== '' && $scope.reportName && $scope.reportName!=='' && $scope.startDate
         && $scope.startDate!=='' && $scope.selectedIndicatorTags.indicatorTags &&
         $scope.selectedIndicatorTags.indicatorTags !== [])
-        var locations =getSelectedLocations($scope.selectedLocations);
+        var locations = '';
+      if ($scope.selectedLocation) {
+        locations = $scope.selectedLocation;
```

Comment:
if $scope.selectedLocation is an empty string (i.e ' ') this if will pass. You have substituted the earlier problem for this one. you might write it like below:
`if($scope.selectedLocation && $scope.selectedLocation !== '') {
   // Do your thing here
}`